app/services/unsubscribe_service.rb

```diff
@@ -8,14 +8,22 @@ def subscription!(subscription, reason)
       unsubscribe!(subscription.subscriber, [subscription], reason)
     end
 
+    def spam_report!(delivery_attempt)
+      subscriber_id = delivery_attempt.email.subscriber_id
+      subscriber = Subscriber.find(subscriber_id)
+      unsubscribe!(subscriber, subscriber.active_subscriptions, :marked_as_spam, delivery_attempt.email)
+    end
+
   private
 
-    def unsubscribe!(subscriber, subscriptions, reason)
+    def unsubscribe!(subscriber, subscriptions, reason, email = nil)
       ActiveRecord::Base.transaction do
         subscriptions.each do |subscription|
           subscription.end(reason: reason)
         end
 
+        email.update!(marked_as_spam: true) if email
```

Comment:
This feels a little non-obvious to me - with the diff in front of me I can see that we only pass the `email` in if we have a spam report, but I wonder if your future selves will notice that or if we'll just get confused. Once we've added the extra argument in, we might be tempted to use it in the future and wouldn't expect the presence of an `email` to trigger a `marked_as_spam` from the method definition alone. Could we make the `if` more explicit perhaps with an `&& reason == :marked_as_spam` or somesuch?

Reply:
Sure, that sounds clearer. Amended now.